### PR TITLE
Update call to read_input_nml and remove unnecessary code.

### PR DESCRIPTION
**Pull request by laurenchilutti**
**Description**

When reviewing PR #136 Rusty and I determined that we should call ```read_input_nml``` with the specifier "alt_input_nml_path" and use the full filename when reading an input namelist.  I removed a section of code that would define ```Atm(this_grid)%nml_filename``` as nest.## (e.g. nest.02).  Lines 380 - 388 (see code snippet below) are sufficient at defining ```Atm%nml_filename``` with the full filename that I now pass to the FMS function ```read_input_nml```

https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/a3ac97321e7435484f8d830fea356678cf5f8503/model/fv_control.F90#L380-L388

I also have removed an unused input argument in test_cases.F90 ```read_namelist_test_cases``` and removed any use of ```#ifdef INTERNAL_FILE_NML``` as all models need to use INTERNAL_FILE_NML and we no longer need a conditional statement for this.

Fixes # (issue)

**How Has This Been Tested?**

Tested with SHiELD with a C48n4, Regional 3KM and C48 no nests tests.  Answers reproduced.

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: model/fv_control.F90 [hunk 1]
```diff
@@ -445,22 +445,12 @@ subroutine fv_control_init(Atm, dt_atmos, this_grid, grids_on_this_pe, p_split)
...
-     call read_input_nml(Atm(this_grid)%nml_filename) !re-reads into internal namelist
-#endif
+     call read_input_nml(alt_input_nml_path=Atm(this_grid)%nml_filename) !re-reads into internal namelist
      call read_namelist_fv_grid_nml
      call read_namelist_fv_core_nml(Atm(this_grid)) ! do options processing here too?
-     call read_namelist_test_case_nml(Atm(this_grid)%nml_filename)
+     call read_namelist_test_case_nml
      !TODO test_case_nml moved to test_cases
-     call read_namelist_test_case_nml(Atm(this_grid)%nml_filename)
+     call read_namelist_test_case_nml
```
Comment (bensonr) [inline model/fv_control.F90:453]: I don't think we need to do this twice - do we?

**Comment by laurenchilutti** [comment]
I have merged the main branch into this and this is now ready to be merged once reviewed.  I have tested with some SHiELD test cases and answers reproduce.

File: model/fv_control.F90 (hunk 1 above)
Comment (laurenchilutti) [reply model/fv_control.F90:453]: you're right - I don't see any reason that we need to read a namelist twice in a row

File: model/fv_control.F90 [hunk 2]
```diff
@@ -445,22 +445,11 @@ subroutine fv_control_init(Atm, dt_atmos, this_grid, grids_on_this_pe, p_split)
...
-     else
-        Atm(this_grid)%nml_filename = ''
-     endif
-     call read_input_nml(Atm(this_grid)%nml_filename) !re-reads into internal namelist
-#endif
+     call read_input_nml(alt_input_nml_path=Atm(this_grid)%nml_filename) !re-reads into internal namelist
      call read_namelist_fv_grid_nml
      call read_namelist_fv_core_nml(Atm(this_grid)) ! do options processing here too?
-     call read_namelist_test_case_nml(Atm(this_grid)%nml_filename)
      !TODO test_case_nml moved to test_cases
```
Comment (bensonr) [inline model/fv_control.F90:451]: Sorry to be picky here, but I don't think we need this TODO comment anymore.

Comment (laurenchilutti) [reply model/fv_control.F90:451]: I don't mind the extensive review - I want to draw attention to these other lines of code in fv_control.  Would you like me to also remove these comments? 

https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/29396d396bcf57e1121f16d90f29c2d85903ed49/model/fv_control.F90#L57-L58
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/29396d396bcf57e1121f16d90f29c2d85903ed49/model/fv_control.F90#L1053

Comment (bensonr) [reply model/fv_control.F90:451]: It makes sense to clean those up too.

**Comment by laurenchilutti** [comment]
@lharris4 @linjiongzhou I would like to merge this PR in prior to making our next release.  Could you please review this? Thank you!

**Review by lharris4 — APPROVED** [review]
This is a welcome update. Thanks.

**Review by linjiongzhou — APPROVED** [review]

**Review by linjiongzhou — APPROVED** [review]
They look good to me. Thank you!

**Review by bensonr — APPROVED** [review]

**Pull request merged**